ClientApp/src/pages/Blog.js

```diff
@@ -0,0 +1,82 @@
+
+import '../styles/components/blog.scss'
+import Layout from "../components/Layout";
+
+const Blog = props => {
+    const history = useHistory();
+    const blogs = useSelector(state => state.blogs.all);
+    const blog = blogs.find(temp => parseInt(temp.id, 10) === parseInt(props.match.params.id))
+    const [liked, setLiked] = useState(false)
+    if (blog === undefined) {
+        history.replace('/browse');
+        return null
+    }
+    const like_or_dislike = () => {
+        setLiked(prev => !prev)
+    }
+
+    return (
+        <Layout>
+            <div className="m-auto wrapper">
+                <h1 className="title">{blog.title}</h1>
+                <Row xs="1">
+                    <Col md="7">
+                        <section>
+                            {blog.body}
+                        </section>
+                        <h1 className="subtitle">My Subtitle</h1>
+                        <section>
+                            {blog.body}
+                        </section>
+                    </Col>
+                    <Col md="5">
+                        <section className="profile-card">
+                            <div className="profile-card-avatar-and-info">
+                                <img
+                                src="https://www.w3schools.com/howto/img_avatar.png"
+                                alt="avatar"
+                                />
+                                <div className="profile-card-info">
+                                    <span>John Texas</span>
+                                    <div>
+                                        <span>24 blogs</span>
+                                        <i
+                                        className="fas fa-newspaper"
+                                        ></i>
+                                        <span>420 likes</span>
+                                        <i
+                                        className="fas fa-thumbs-up"
+                                        ></i>
+                                    </div>
+                                </div>
+                            </div>
+                            <div className="profile-card-btns">
+                                <button onClick={like_or_dislike}>
+                                    {liked ? 'Liked' : (
+                                        <>
+                                        <span>Like this blog</span>
+                                    <i
+                                    className="fas fa-thumbs-up"
+                                    ></i>
+                                    </>
+                                    )
+                                }
+                                </button>
+                                <br />
+                                <button>
+                                    <span>View John's other blogs</span>
+                                </button>
```

Comment:
Just surrond this in a NavLink as when the time comes, this will link to /profile#blogs